modules/@apostrophecms/attachment/index.js

```diff
@@ -308,29 +317,34 @@ module.exports = {
           silent: silent
         });
       },
-      acceptableExtension(field, attachment) {
-        const groups = field.fileGroups || (field.fileGroup && [ field.fileGroup ]);
+      checkExtension(field, attachment) {
+        const groups = field.fileGroups ||
+          (field.fileGroup && [ field.fileGroup ]);
         let extensions;
+
         if (groups) {
           if (!_.includes(groups, attachment.group)) {
             extensions = [];
             _.each(groups, function (group) {
-              const groupInfo = _.find(self.options.fileGroups, { name: group });
+              const groupInfo = _.find(self.fileGroups, { name: group });
               if (!groupInfo) {
-                return;
+                return [];
               }
               extensions = extensions.concat(groupInfo.extensions);
             });
-            return false;
+
+            return extensions;
           }
         }
-        extensions = field.extensions || (field.extension && [ field.extension ]);
+        extensions = field.extensions ||
+          (attachment.extension && [ attachment.extension ]);
```

Comment:
Fix here. Please confirm.

Reply:
attachment.extension has no business on a list of what's allowed; it's what was actually sent.

The idea here was to support attachment fields specifying a single extension to allow or more than one, as an alternative to specifying a group. The existing code seems correct for that.

Reply:
A method that returns either `true` or an array of things that would have been OK is unusual, so please document it.

Reply:
True. For some reason that seemed necessary, but the `self.options.fileGroups` fix was probably the important one.